### PR TITLE
Ps 142 posts can be deleted

### PR DESCRIPTION
**Pull request by AlexDjangoX**
Schema updated for cascading deletion of comments and posts.

User can delete their own posts, and associated comments.

Issue #142 Posts can be deleted.

### CODE REVIEW

File: src/routes/post.js [hunk 1]
```diff
@@ -13,5 +14,6 @@ router.post('/', validateAuthentication, create)
 router.get('/', validateAuthentication, getAll)
 router.post('/comment', validateAuthentication, createComment)
 router.get('/comment', validateAuthentication, findAllComments)
+router.delete('/post-delete/:id', validateAuthentication, deletePost)
```
Comment (dearshrewdwit) [inline src/routes/post.js:17]: Please change this to `/posts/:id` follow a conventional RESTful endpoint

**Review by dearshrewdwit — CHANGES_REQUESTED** [review]

File: src/controllers/user.js [hunk 2]
```diff
@@ -1,5 +1,5 @@
 import User from '../domain/user.js'
-import Cohort from '../domain/cohort.js'
```
[the remark this answers is not in the document]
Comment (AlexDjangoX) [reply src/controllers/user.js:2]: 'Cohort' is declared but its value is never read.

File: src/domain/cohort.js [hunk 3]
```diff
@@ -46,22 +46,6 @@ export default class Cohort {
     this.name = name
   }
 
-  static async findCohortByID(id) {
```
[the remark this answers is not in the document]
Comment (AlexDjangoX) [reply src/domain/cohort.js:49]: Random line of code with no body or closing parenthesis

File: src/routes/post.js (hunk 1 above)
Comment (AlexDjangoX) [reply src/routes/post.js:17]: Changed.

File: src/controllers/post.js [hunk 4]
```diff
@@ -61,3 +61,14 @@ export const getAll = async (req, res) => {
     return sendDataResponse(res, 400, { err: err.message })
   }
 }
+
+export const deletePost = async (req, res) => {
+  const postId = +req.params.id
+  if (!postId) throw new Error('The ID you have provided is incorrect')
```
Comment (dearshrewdwit) [inline src/controllers/post.js:67]: Catch this error - moving into the try in this function, or adding it in the parent function

**Review by dearshrewdwit — CHANGES_REQUESTED** [review]

File: src/controllers/user.js [hunk 5]
```diff
@@ -47,21 +47,13 @@ export const getById = async (req, res) => {
...
-  } = req.query
+  const { first_name: firstName, cohort: inCohort } = req.query
 
   const whereData = {}
   if (inCohort === 'false') {
     whereData.cohort = null
   }
 
-  if (cohortId) {
-    whereData.cohortId = Number(cohortId)
```
Comment (dearshrewdwit) [inline src/controllers/user.js:62]: Checking - is this no longer needed?

**Review by dearshrewdwit — CHANGES_REQUESTED** [review]
Great - looking cleaner.

1 check for the remaining deletion.

**Review by SeanRichhh — APPROVED** [review]

File: src/controllers/user.js (hunk 5 above)
Comment (AlexDjangoX) [reply src/controllers/user.js:62]: Sean pulled from main and that code is not in main.

Also if i add it, it says cohortId is not defined.

**Review by Auenc — APPROVED** [review]
LGTM

**Review by SeanRichhh — APPROVED** [review]

**Pull request merged**